src/networkapi.jl

```diff
@@ -183,6 +183,135 @@ function netstoichmat(rn; smap=speciesmap(rn))
 end
 
 
+######################## some reacion complexes matrices and reaction rates ###############################
+"""
+reaction_complexes(rn)
+The function returns a matrix of unique complexes
+    that represent the reaction network,
+reaction_complexes(rn)[:,1] returns Vector of
+complexes and corresponding stoichiometric coefficient
+
+reaction_complexes(rn)[:,2] returns a vector of linear combination of complexes(coefficients being corresponding stoichiometric coefficient)
```

Comment:
> In fact, I wonder if it wouldn't make more sense to just return the species (integer) indices and the numeric coefficients, as vectors of vectors of pairs (where each pair corresponds to a species index and a stoichiometric coefficient), i.e. don't return anything symbolic. Generally I think it is better for these routines to work with species identified by integers and stoichiometric coefficients. This will be less memory intensive and more performant for large systems.

@isaacsas , i thought about this too. How do you recommend we should denote `∅` type of complex ? 
Empty array for index and zero for stoichiometric coefficient ??

Reply:
Small demo here for this..

```julia
    sp = species(rn);
    r = reactions(rn);
    complexes_mat = []
    for i in 1:numreactions(rn)
        push!(complexes_mat,Pair(indexin(r[i].substrates,sp) , r[i].substoich));
        push!(complexes_mat,Pair(indexin(r[i].products,sp) , r[i].prodstoich))

    end
    unique(complexes_mat)
```

Reply:
Maybe pass in `speciesmap` as an optional keyword arg, and use that instead of `indexin`? It should be faster for big systems. Also, you can use `unique!`

This new form of `complexes_mat` looks good, or alternatively just return two vectors of vectors. One with the species and one with the stoichiometry.

Finally, you should try to give `complexes_mat` a concrete type instead of `Any`. i.e. `Vector{Pair{...}}` if possible.

Reply:
hey @isaacsas , i have updated https://github.com/SciML/Catalyst.jl/pull/359#discussion_r680366630 , and other functions too depending on this new way(check in other comments below). Let me know what you think